boards.txt

```diff
@@ -16261,6 +16373,371 @@ esp32c6-evb.menu.EraseFlash.all.upload.erase_cmd=-e
+esp32h2-devkitlipo.menu.ZigbeeMode.ed=Zigbee ED (end device)
+esp32h2-devkitlipo.menu.ZigbeeMode.ed.build.zigbee_mode=-DZIGBEE_MODE_ED
+esp32h2-devkitlipo.menu.ZigbeeMode.ed.build.zigbee_libs=-lesp_zb_api_ed -lesp_zb_cli_command -lzboss_stack.ed.trace -lzboss_stack.ed -lzboss_port
+esp32h2-devkitlipo.menu.ZigbeeMode.zczr=Zigbee ZCZR (coordinator)
+esp32h2-devkitlipo.menu.ZigbeeMode.zczr.build.zigbee_mode=-DZIGBEE_MODE_ZCZR
+esp32h2-devkitlipo.menu.ZigbeeMode.zczr.build.zigbee_libs=-lesp_zb_api_zczr -lesp_zb_cli_command -lzboss_stack.zczr.trace -lzboss_stack.zczr -lzboss_port
+esp32h2-devkitlipo.menu.ZigbeeMode.rcp=Zigbee RCP (radio co-processor)
+esp32h2-devkitlipo.menu.ZigbeeMode.rcp.build.zigbee_mode=-DZIGBEE_MODE_RCP
+esp32h2-devkitlipo.menu.ZigbeeMode.rcp.build.zigbee_libs=-lesp_zb_api_rcp -lesp_zb_cli_command -lzboss_stack.rcp -lzboss_port
+
+##############################################################
+
+esp32-sbc-fabgl.name=OLIMEX ESP32-SBC-FABGL
+
+esp32-sbc-fabgl.bootloader.tool=esptool_py
+esp32-sbc-fabgl.bootloader.tool.default=esptool_py
+
+esp32-sbc-fabgl.upload.tool=esptool_py
+esp32-sbc-fabgl.upload.tool.default=esptool_py
+esp32-sbc-fabgl.upload.tool.network=esp_ota
+
+esp32-sbc-fabgl.upload.maximum_size=1310720
+esp32-sbc-fabgl.upload.maximum_data_size=327680
+esp32-sbc-fabgl.upload.flags=
+esp32-sbc-fabgl.upload.extra_flags=
+
+esp32-sbc-fabgl.serial.disableDTR=true
+esp32-sbc-fabgl.serial.disableRTS=true
+
+esp32-sbc-fabgl.build.tarch=xtensa
+esp32-sbc-fabgl.build.bootloader_addr=0x1000
+esp32-sbc-fabgl.build.target=esp32
+esp32-sbc-fabgl.build.mcu=esp32
+esp32-sbc-fabgl.build.core=esp32
+esp32-sbc-fabgl.build.variant=esp32-sbc-fabgl
+esp32-sbc-fabgl.build.board=ESP32_SBC_FABGL
+
+esp32-sbc-fabgl.build.f_cpu=240000000L
+esp32-sbc-fabgl.build.flash_size=4MB
+esp32-sbc-fabgl.build.flash_freq=40m
+esp32-sbc-fabgl.build.flash_mode=dio
+esp32-sbc-fabgl.build.boot=dio
+esp32-sbc-fabgl.build.partitions=default
+esp32-sbc-fabgl.build.defines=
+esp32-sbc-fabgl.build.loop_core=
+esp32-sbc-fabgl.build.event_core=
+
+## IDE 2.0 Seems to not update the value
+esp32-sbc-fabgl.menu.JTAGAdapter.default=Disabled
+esp32-sbc-fabgl.menu.JTAGAdapter.default.build.copy_jtag_files=0
+esp32-sbc-fabgl.menu.JTAGAdapter.external=FTDI Adapter
+esp32-sbc-fabgl.menu.JTAGAdapter.external.build.openocdscript=esp32-wrover-kit-3.3v.cfg
+esp32-sbc-fabgl.menu.JTAGAdapter.external.build.copy_jtag_files=1
+esp32-sbc-fabgl.menu.JTAGAdapter.bridge=ESP USB Bridge
+esp32-sbc-fabgl.menu.JTAGAdapter.bridge.build.openocdscript=esp32-bridge.cfg
+esp32-sbc-fabgl.menu.JTAGAdapter.bridge.build.copy_jtag_files=1
+
+esp32-sbc-fabgl.menu.PSRAM.disabled=Disabled
+esp32-sbc-fabgl.menu.PSRAM.disabled.build.defines=
+esp32-sbc-fabgl.menu.PSRAM.disabled.build.extra_libs=
+esp32-sbc-fabgl.menu.PSRAM.enabled=Enabled
+esp32-sbc-fabgl.menu.PSRAM.enabled.build.defines=-DBOARD_HAS_PSRAM -mfix-esp32-psram-cache-issue -mfix-esp32-psram-cache-strategy=memw
```

Comment:
Psram fix not needed. Can be removed.

Reply:
Hello @Jason2866 and thanks for the suggestion.

When I created the board entry in the board.txt I just copied that part from the generic ESP32 board (lines 1131-1318) because I have tested that it worked with it being selected. And after that I just changed the name name, build.variant and build.board.
I wasn't aware of what these mfix options does so I thought they are there for a reason and I just let them stay there.
If you are suggesting that they are unnecessary and can be safely removed that's all good.

The question I have is should I remove them from my forked repository and make a new pull request or you can do it directly on your repository?

Reply:
Dont have the needed rights to change. Not a member -> Collaborator

Reply:
@Stanimir-Petev just remove `-mfix-esp32-psram-cache-issue -mfix-esp32-psram-cache-strategy=memw` from all non-ESP32 boards. Those flags are valid only on the original ESP32

Reply:
Sorry on my end... @Jason2866 this is ESP32 based board. Why did you ask the flags to be removed?

Reply:
@me-no-dev Since it is clearly documented that **only** old revisions of the esp32 do need this fix. https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/external-ram.html#chip-revisions

Reply:
Sadly many boards have this absolutely not needed entry, since the are newer than revision 1.

Reply:
@me-no-dev We are checking in project Tasmota the esp32 revision and search for PSRAM at init phase and enable it when revision >1. This works perfect.
When the revision check identifies esp32 with the bug PSRAM is not enabled and a message is printed, for PSRAM usage the fix files needs to be used during compile.

Reply:
@Jason2866 so how do you detect that here? Do we know for sure that all boards are ECO3?

Reply:
@me-no-dev Just recommended to remove from the *new* board "esp32-sbc-fabgl"

Reply:
@Stanimir-Petev can you confirm that you are using only V3 ESP32 modules?

Reply:
If that is true, then I can agree, but then why is PSRAM not enabled by default?

Reply:
True, PSRAM should be always enabled!

Reply:
No way to confirm if all ESP32-POE-WROVER and ESP32-POE-ISO-WROVER variants had version 3 modules. But these were made starting 2020 so good chance they are all version 3.

If you ask for all ESP32-POE and ESP32-POE-ISO boards - these use WROOM modules and had been in production since the early days of the ESP32.

We would enable the PSRAM for the ESP32-SBC_FABGL, for the rest the variants with WROOM module are still more popular and it makes more sense to have the PSRAM disabled. We don't want to create addition entries for each variant of the board, since some boards would need to have 3 entries like ESP32-POE, ESP32-POE-WROVER, ESP32-POE-16MB all the same design with different module soldered (one with WROVER module and PSRAM, and the one with bigger flash).

Reply:
With the latest commit PSRAM enabled should be the default option on ESP32-SBC-FabGL. Is there anything else that has to be done?